front/app/components/HookForm/OptionList/index.tsx

```diff
@@ -0,0 +1,277 @@
+import React, { useState, useCallback } from 'react';
+import { DndProvider } from 'react-dnd-cjs';
+import HTML5Backend from 'react-dnd-html5-backend-cjs';
+import { get } from 'lodash-es';
+
+// react hook form
+import { Controller, useFieldArray, useFormContext } from 'react-hook-form';
+
+// components
+import {
+  Box,
+  Label,
+  Button,
+  LocaleSwitcher,
+  Icon,
+  Input,
+} from '@citizenlab/cl2-component-library';
+import { SectionField } from 'components/admin/Section';
+import { List, SortableRow } from 'components/admin/ResourceList';
+import Error, { TFieldName } from 'components/UI/Error';
+
+// Typings
+import { Locale, CLError, RHFErrors, Multiloc } from 'typings';
+
+// utils
+import { isNilOrError } from 'utils/helperUtils';
+import { generateTempId } from 'components/FormBuilder/components/FormBuilderSettings/utils';
+
+type Option = {
+  id?: string;
+  temp_id?: string;
+  title_multiloc: Multiloc;
+};
+
+interface Props {
+  name: string;
+  onSelectedLocaleChange?: (locale: Locale) => void;
+  locales: Locale[];
+  allowDeletingAllOptions?: boolean;
+  platformLocale: Locale;
+  fieldLabel: string | JSX.Element | null;
+  addButtonLabel: string | JSX.Element | undefined;
+}
+
+const OptionList = ({
+  onSelectedLocaleChange,
+  name,
+  locales,
+  allowDeletingAllOptions = false,
+  platformLocale,
+  fieldLabel,
+  addButtonLabel,
+}: Props) => {
+  const {
+    control,
+    formState: { errors: formContextErrors },
+    setValue,
+    trigger,
+  } = useFormContext();
+  const [selectedLocale, setSelectedLocale] = useState<Locale | null>(
+    platformLocale
+  );
+
+  // Handles locale change
+  const handleOnSelectedLocaleChange = useCallback(
+    (newSelectedLocale: Locale) => {
+      setSelectedLocale(newSelectedLocale);
+      onSelectedLocaleChange?.(newSelectedLocale);
+    },
+    [onSelectedLocaleChange]
+  );
+
+  // Handle option change
+  type HandleOptionChangeProps = {
+    value: string;
+    options: Array<Option>;
+    index: number;
+    name: string;
+    selectedLocale: Locale;
+  };
+
+  const handleOptionChange = ({
+    value,
+    options,
+    index,
+    name,
+    selectedLocale,
+  }: HandleOptionChangeProps) => {
+    const updatedOptions = options;
```

Comment:
```suggestion
    const updatedOptions = cloneDeep(options);
```
It's usually not good practice to mutate arguments passed to a function like this